lib/view_component/base.rb

```diff
@@ -636,6 +636,11 @@ def iteration_argument_present?
 
       private
 
+      def splatted_keyword_argument_present?
+        initialize_parameters.flatten.include?(:keyrest) &&
+          !initialize_parameters.include?([:keyrest, :**]) # Un-named splatted keyword args don't count!
```

Comment:
Is there a specific reason for not accepting unnamed splatted arguments? This limitation creates some problems with initializers, which forward their arguments to helpers, e.g. dry-initialize, which uses `def initialize(...); __dry_initializer_initialize__(...); end` (`...` = actual Ruby code) or any kind of custom helper, like `def initialize(**); do_init(**); end;` (Sorry for "gravedigging", I'll open a separate issue, if that is more appropriate...)

Reply:
I can't remember my exact reasoning, but it was more a limitation-in-my-capacity and not necessarily a limitation-by-design.

Reply:
@daniel-rikowski @zspencer can we please open up a new issue where this can be discussed further

Reply:
👋 I'd also be interested in a PR with a failing test for the use case you want 😄

Reply:
See #2020

@zspencer It seems this was because of a Ruby 3.1 bug 😉